src/AmqpBundle/Amqp/Producer.php

```diff
@@ -50,14 +50,15 @@ public function publishMessage($message, $flags = AMQP_NOPARAM, array $attribute
                       (empty($this->exchangeOptions['publish_attributes']) ? $attributes :
                       array_merge($this->exchangeOptions['publish_attributes'], $attributes));
 
-        if (empty($routingKeys)) {
-            $routingKeys = $this->exchangeOptions['routing_keys'];
+        $routingKeys = !empty($routingKeys) ? $routingKeys : $this->exchangeOptions['routing_keys'];
+        if (!$routingKeys) {
+            return $this->call($this->exchange, 'publish', [$message, null, $flags, $attributes]);
         }
 
         // Publish the message for each routing keys
-        $success = true;
+        $success = false;
         foreach ($routingKeys as $routingKey) {
-            $success &= $this->call($this->exchange, 'publish', [$message, $routingKey, $flags, $attributes]);
+            $success = $this->call($this->exchange, 'publish', [$message, $routingKey, $flags, $attributes]);
```

Comment:
with two routing keys, if the first "call" failed and the second is ok, $success will be set to true, and it's not what you what no ?